python/instrumentation/openinference-instrumentation-boto/src/openinference/instrumentation/boto/__init__.py

```diff
@@ -0,0 +1,169 @@
+                response["body"] = BufferedStreamingBody(
+                    response["body"]._raw_stream, response["body"]._content_length
+                )
+                if raw_request_body := kwargs.get("body"):
+                    request_body = json.loads(raw_request_body)
+                response_body = json.loads(response.get("body").read())
+                response["body"].reset()
+
+                prompt = request_body.pop("prompt")
+                invocation_parameters = request_body
+
+                _set_span_attribute(span, SpanAttributes.LLM_PROMPTS, prompt)
+                _set_span_attribute(
+                    span, SpanAttributes.LLM_INVOCATION_PARAMETERS, invocation_parameters
+                )
+
+                if model_id := kwargs.get("modelId"):
+                    _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+
+                    if isinstance(model_id, str):
+                        (vendor, _) = model_id.split(".")
+
+                    if vendor == "ai21":
+                        content = str(response_body.get("completions"))
+                        _set_span_attribute(
+                            span,
+                            MessageAttributes.MESSAGE_CONTENT,
+                            content,
+                        )
+                    elif vendor == "anthropic":
+                        content = str(response_body.get("completion"))
+                        _set_span_attribute(span, MessageAttributes.MESSAGE_CONTENT, content)
+                    elif vendor == "cohere":
+                        content = str(response_body.get("generations"))
+                        _set_span_attribute(
+                            span,
+                            MessageAttributes.MESSAGE_CONTENT,
+                            content,
+                        )
+
+                return response  # type: ignore
+
+        return instrumented_response
+
+    return _invocation_wrapper
+
+
+class BotoInstrumentor(BaseInstrumentor):  # type: ignore
+    __slots__ = ("_original_client_creator",)
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs: Any) -> None:
+        if not (tracer_provider := kwargs.get("tracer_provider")):
+            tracer_provider = trace_api.get_tracer_provider()
+        tracer = trace_api.get_tracer(__name__, __version__, tracer_provider)
+
+        boto = import_module(_MODULE)
+        self._original_client_creator = boto.ClientCreator.create_client
+
+        wrap_function_wrapper(
+            module=_MODULE,
+            name="ClientCreator.create_client",
+            wrapper=_client_creation_wrapper(tracer=tracer),
+        )
+
+    def _uninstrument(self, **kwargs: Any) -> None:
+        boto = import_module(_MODULE)
+        boto.ClientCreator.create_client = self._original_client_creator
```

Comment:
nit. clean up the pointer
```suggestion
        boto.ClientCreator.create_client = self._original_client_creator
        self._original_client_creator = None
```